File: domain-driven-design/java/repository/OrderRepository.java
// repository/OrderRepository.java
package repository;

import domain.Order;
import java.util.HashMap;
import java.util.Map;

public class OrderRepository {
    private Map<Integer, Order> orders = new HashMap<>();

    public void save(Order order) {
        orders.put(order.getId(), order);
    }

    public Order findById(int id) {
        return orders.get(id);
    }

    public void findAll() {
        if (orders.isEmpty()) {
            System.out.println("暂无订单历史记录。");
        } else {
            for (Order order : orders.values()) {
                order.display();
            }
        }
    }

    public void clear() {
        orders.clear();
        System.out.println("所有订单已清理");
    }
}

File: domain-driven-design/js/repository/order_repository.js
// 仓储层，负责与数据源进行交互
// repository/order_repository.js
export class OrderRepository {
    constructor() {
        this.orders = new Map();
    }

    save(order) {
        this.orders.set(order.id, order);
    }

    findById(id) {
        return this.orders.get(id) || null;
    }

    findAll() {
        return Array.from(this.orders.values());
    }

    clear() {
        this.orders.clear();
        console.log("所有订单已清理");
    }
}